src/validators/relatives.js

```diff
@@ -18,11 +27,40 @@ export default class RelativesValidator {
     })
   }
 
+  validMaritalRelations(context = null) {
+    const maritalStatuses = ['Married', 'Separated']
+    const requiredRelations = ['Father-in-law', 'Mother-in-law']
+    const maritalStatus = context || getContext()
+
+    console.log(maritalStatus)
+
+    if (!maritalStatus || !maritalStatuses.includes(maritalStatus)) {
+      return true
+    }
+
+    if ((this.list.branch || {}).value === 'No') {
+      if (this.list.items || {}.length > 0) {
+        let relations = []
+        for (const item of this.list.items) {
+          if (!item || !item.Item || !item.Item.Relation) {
+            continue
+          }
+          relations.push(item.Item.Relation.value)
+        }
+        if (relations.length > 0) {
+          return requiredRelations.every(r => relations.includes(r))
+        }
+      }
+      return false
+    }
+    return true
+  }
+
   validMinimumRelations() {
     const requiredRelations = ['Father', 'Mother']
 
     if ((this.list.branch || {}).value === 'No') {
-      if (this.list.items && this.list.items.length > 0) {
+      if (this.list.items || {}.length > 0) {
```

Comment:
This should get enclosing parens before calling length?

Reply:
Yeah, updated, thanks.